lib/ansible/modules/system/user.py

```diff
@@ -1442,11 +1442,23 @@ def create_user(self):
                         fields[1] = self.password
                         fields[2] = str(int(time.time() // 86400))
                         if minweeks:
-                            fields[3] = str(int(minweeks) * 7)
+                            try:
+                                fields[3] = str(int(minweeks) * 7)
+                            except ValueError:
+                                # mirror solaris, which allows for any value in this field, and ignores anything that is not an int.
+                                pass
                         if maxweeks:
-                            fields[4] = str(int(maxweeks) * 7)
+                            try:
+                                fields[4] = str(int(maxweeks) * 7)
+                            except ValueError:
+                                # mirror solaris, which allows for any value in this field, and ignores anything that is not an int.
+                                pass
                         if warnweeks:
-                            fields[5] = str(int(warnweeks) * 7)
+                            try:
+                                fields[5] = str(int(warnweeks) * 7)
+                            except ValueError:
+                                # mirror solaris, which allows for any value in this field, and ignores anything that is not an int.
+                                pass
```

Comment:
Hi,
just small comment on this. In function `def modify_user_usermod(self):` there is the same code which also should be surround with `try, except` clause:
```python
                     for line in open(self.SHADOWFILE, 'rb').readlines():
                         line = to_native(line, errors='surrogate_or_strict')
                         fields = line.strip().split(':')
                         if not fields[0] == self.name:
                             lines.append(line)
                             continue
                         fields[1] = self.password
                         fields[2] = str(int(time.time() // 86400))
                         if minweeks:
                             fields[3] = str(int(minweeks) * 7)
                         if maxweeks:
                             fields[4] = str(int(maxweeks) * 7)
                         if warnweeks:
                             fields[5] = str(int(warnweeks) * 7)
                         line = ':'.join(fields)
                         lines.append('%s\n' % line)
```

@bengerman13 can you add this change there as well, please?